chunky/src/java/se/llbit/chunky/world/biome/ArrayBiomePalette.java

```diff
@@ -0,0 +1,61 @@
+package se.llbit.chunky.world.biome;
+
+import se.llbit.log.Log;
+import se.llbit.util.NotNull;
+
+import java.util.List;
+import java.util.Objects;
+
+public class ArrayBiomePalette implements BiomePalette {
+  public static final int ARRAY_BIOMEPALETTE_SIZE = 5;
+
+  private Biome[] palette;
+  private int size = 0;
+
+  public ArrayBiomePalette() {
+    palette = new Biome[ARRAY_BIOMEPALETTE_SIZE];
+  }
+
+  public ArrayBiomePalette(@NotNull List<Biome> palette) {
+    assert palette.stream().noneMatch(Objects::isNull);
+
+    this.palette = palette.toArray(new Biome[0]);
+  }
+
+  @Override
+  public Biome get(int id) {
+    assert id >= 0 && id < size;
+
+    return this.palette[id];
+  }
+
+  @Override
+  public int put(@NotNull Biome biome) {
+    assert biome != null;
+
+    if(size == this.palette.length) {
+      this.grow();
```

Comment:
Why not use an `ArrayList` instead of implementing the grow logic by ourselves?

Reply:
technically there is some time to be saved implementing it ourselves, though it's not very significant. I'll switch to `ArrayList`